File: nba-wins-platform/tasks/season_config.py
"""
Season Configuration Loader for Python Scripts

Loads season dates and parameters from config/season.json.
Update season.json once per year — all files read from it automatically.

Usage:
    from season_config import get_season_config
    season = get_season_config()
    print(season['season_start_date'])  # '2025-10-21'
"""

import json
import os


def get_season_config():
    """Load and return the season configuration from season.json."""
    config_path = os.path.join(
        os.path.dirname(os.path.abspath(__file__)),
        '..', 'config', 'season.json'
    )
    try:
        with open(config_path) as f:
            return json.load(f)
    except (FileNotFoundError, json.JSONDecodeError) as e:
        print(f"Warning: Could not load season config: {e}")
        # Fallback defaults so scripts don't crash
        return {
            'season_label': '2025-26',
            'api_season_nba': '2025-26',
            'api_season_rapid': '2025',
            'api_season_espn': '2026',
            'season_start_date': '2025-10-21',
        }

File: nba-wins-platform/config/season_config.php
<?php
/**
 * Season Configuration Loader
 *
 * Loads season dates and parameters from season.json.
 * Update season.json once per year — all files read from it automatically.
 *
 * Usage:
 *   require_once '/data/www/default/nba-wins-platform/config/season_config.php';
 *   $season = getSeasonConfig();
 *   echo $season['season_start_date'];  // '2025-10-21'
 */

function getSeasonConfig() {
    static $config = null;
    if ($config === null) {
        $json = file_get_contents(__DIR__ . '/season.json');
        $config = json_decode($json, true);
        if ($config === null) {
            error_log("Failed to parse season.json: " . json_last_error_msg());
            // Fallback defaults so the site doesn't crash
            $config = [
                'season_label' => '2025-26',
                'standings_table' => '2025_2026',
                'standings_table_backup' => '2025_2026_backup',
                'season_start_date' => '2025-10-21',
                'api_season_nba' => '2025-26',
                'api_season_rapid' => '2025',
                'api_season_espn' => '2026',
                'all_star_break_start' => '2026-02-13',
                'all_star_break_end' => '2026-02-18',
                'nba_cup_dates' => [],
            ];
        }
    }
    return $config;
}
